flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/optimize/StreamCommonSubGraphBasedOptimizer.scala

```diff
@@ -46,62 +47,98 @@ import scala.collection.JavaConversions._
 class StreamCommonSubGraphBasedOptimizer(planner: StreamPlanner)
   extends CommonSubGraphBasedOptimizer {
 
+  private def shouldSkipMiniBatch(blocks: Seq[RelNodeBlock]): Boolean = {
```

Comment:
I choose to put it into a util to avoid introducing specific nodes such as filter, project, etc within `StreamCommonSubGraphBasedOptimizer`. WDYT? 0.0